src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -348,9 +348,7 @@ public boolean authenticate(final SecurityRequestChannel request) {
                         // saml will always hit this to re-request authentication
                         if (!authDomain.getHttpAuthenticator().getType().equals(SAML_TYPE)) {
                             auditLog.logFailedLogin("<NONE>", false, null, request);
-                        }
-                        if (isTraceEnabled) {
-                            log.trace("No 'Authorization' header, send 401 and 'WWW-Authenticate Basic'");
+                            log.warn("No 'Authorization' header, send 401 and 'WWW-Authenticate Basic'");
```

Comment:
qq: wouldn't this now flood the logs for other auth types with a warning log? 

https://github.com/opensearch-project/security/blob/0d10f9eae76b0e280d26201bf15f65cec8361aee/src/main/java/org/opensearch/security/securityconf/impl/v7/ConfigV7.java#L315

Reply:
while it makes sense for SAML/basic auth, I think it's set as ``true`` as default for all other auth types as well?
https://github.com/opensearch-project/security/blob/0d10f9eae76b0e280d26201bf15f65cec8361aee/src/main/java/org/opensearch/security/securityconf/impl/v7/ConfigV7.java#L315

https://github.com/opensearch-project/security/blob/0d10f9eae76b0e280d26201bf15f65cec8361aee/src/main/java/org/opensearch/security/securityconf/DynamicConfigModelV7.java#L347-L351


Also, security documentation suggests challenge to be ``true`` for LDAP as well- https://docs.opensearch.org/docs/latest/security/authentication-backends/ldap/#use-active-directory-and-ldap-for-authentication

I think an auth-type based check to log the line makes sense

Reply:
Added a check to only log this on Basic auth. wdyt of this PR now?